From 43d5103dee4836d5bb4a72a3c5298f00f7571af4 Mon Sep 17 00:00:00 2001
From: Ang Tien Sung <tien.sung.ang@intel.com>
Date: Fri, 10 Dec 2021 11:32:26 +0800
Subject: [PATCH] HSD #15010184463: crypto: intel_fcs: Fix the inability to
 probe driver on S10 DTSI

commit 85d7cb262602855c4f22ff5e96b14d6147d1ab80 from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Properly define fcs device in the device tree and add proper driver

Signed-off-by: Ang Tien Sung <tien.sung.ang@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
index dd3983ec6cd6..3d0d27e90b0b 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
@@ -616,6 +616,11 @@
 				fpga_mgr: fpga-mgr {
 					compatible = "intel,stratix10-soc-fpga-mgr";
 				};
+
+				fcs: fcs {
+					compatible = "intel,agilex-soc-fcs";
+				};
+
 			};
 		};
 	};
-- 
2.34.1

